app/base_robot.py

```diff
@@ -0,0 +1,37 @@
+from app.constants import X_INDEX, Y_INDEX, BASE_ROBOT_STEP
+
+
+def list_isinstance(checking: list | None, value: int) -> int:
+    return checking[value] if isinstance(checking, list) else 0
+
+
+class BaseRobot:
+
+    def __init__(self, name: str, weight: int,
+                 coords: list | None = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.x_pos = list_isinstance(coords, X_INDEX)
+        self.y_pos = list_isinstance(coords, Y_INDEX)
+        self.coords = [self.x_pos, self.y_pos]
+
+    def move(self, coord_name: str, step: int,
+             simbl: str, coord_index: int) -> None:
+        value = eval(f"self.{coord_name} {simbl} {step}")
+        setattr(self, coord_name, value)
+        self.coords[coord_index] = value
```

Comment:
```suggestion
        value = eval(f"self.coords[{coord_index}] {simbl} {step}")
        self.coords[coord_index] = value
```
And totally remove x_pos, y_pos, z_pos just use indexes